### PR TITLE
fix bug test fail in tidb4.0 test

### PR DESCRIPTION
**Pull request by qidi1**
Signed-off-by: qidi1 <1083369179@qq.com>

### What problem does this PR solve? 

close: #2479
### What is changed and how it works?

When TiDB version is smaller than `5.0.0`. The `test duplicate unique indexes are not deleted error` and `test bug fix incorrect uniqueIndex key when table is not intHandle` will not run.

### CODE REVIEW

**Comment by qidi1** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

File: core/src/test/scala/org/apache/spark/sql/insertion/BatchWritePKAndIndexSuite.scala [hunk 1]
```diff
@@ -116,6 +117,9 @@ class BatchWritePKAndIndexSuite
   // https://github.com/pingcap/tispark/issues/2452
   test("test duplicate unique indexes are not deleted error") {
     tidbStmt.execute("drop table if exists `tispark_test`.`t`")
+    if (StoreVersion.minTiKVVersion("5.0.0", this.ti.tiSession.getPDClient)) {
```
Comment (zhangyangyu) [inline core/src/test/scala/org/apache/spark/sql/insertion/BatchWritePKAndIndexSuite.scala:120]: Isn't the logic totally reversed? You need to skip the test if it's smaller than version 5.0.0, but you are skipping it when it's bigger than version 5.0.0.

**Comment by qidi1** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

**Review by zhangyangyu — APPROVED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Comment by zhangyangyu** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

**Comment by qidi1** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

**Review by shiyuhang0 — APPROVED** [review]

**Comment by qidi1** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

**Comment by qidi1** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

**Comment by qidi1** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

**Pull request merged**